numba/tests/test_array_iterators.py

```diff
@@ -470,6 +482,16 @@ def test_iter_next(self):
         arr = np.arange(12, dtype=np.int32) + 10
         self.check_array_unary(arr, typeof(arr), func)
 
+    def test_ndenumerate_zero_dim(self):
+        """Test np.ndenumerate with zero-dimensional arrays (issue #10127)."""
+        # Test the original reproducer from the issue
+        cfunc = njit(_nan_maximum_impl)
+        
+        # This should not fail with IndexError anymore
+        result = cfunc((np.array(1.0),))
+        expected = _nan_maximum_impl((np.array(1.0),))
```

Comment:
This should use the idiom where the `.py_func` on a `@njit` compiled function is used to obtain the `expected` result.